hasher-matcher-actioner/hmalib/fetching/bank_store.py

```diff
@@ -0,0 +1,289 @@
+        # member.
+        unique_content_types = set(
+            # Is there a signal type that may have multiple content_types?
+            list(map(lambda st: st.get_content_types()[0], signal_type_view.keys()))
+        )
+        if len(unique_content_types) > 1:
+            raise MultipleContentTypesInFetchDeltaRow()
+
+        content_type = unique_content_types.pop()
+
+        opinion_lists = [
+            metadata.get_as_opinions()
+            for val in signal_type_view.values()
+            for metadata in val.values()
+        ]
+        unique_opinion_tags = set()
+        for opinion_list in opinion_lists:
+            unique_opinion_tags.update(self.convert_opinions_to_label(opinion_list))
+
+        try:
+            member = self.banks_table.add_keyed_bank_member(
+                self._import_bank.bank_id,
+                str_key,
+                content_type,
+                None,
+                None,
+                None,
+                "",
+                True,
+                unique_opinion_tags,
+            )
+        except KeyError:
+            # member already exists, update labels just to be sure.
+            member = self.banks_table.update_bank_member(
+                self.banks_table._key_for_bank(self._import_bank.bank_id, str_key),
+                notes="",
+                bank_member_tags=unique_opinion_tags,
+            )
+
+        for signal_type in signal_type_view:
+            for hash_value in signal_type_view[signal_type].keys():
+                self.banks_table.add_bank_member_signal(
+                    bank_id=self._import_bank.bank_id,
+                    bank_member_id=member.bank_member_id,
+                    signal_type=signal_type,
+                    signal_value=hash_value,
+                )
+
+                # TODO: We don't verify that all existing bank-member signals should continue existing.
+
+    def _handle_delete(self, key: TUpdateRecordKey):
+        """
+        Respond to a delete from a signal exchange API. Typically, this would
+        happen at most once per key per flush.
+        """
+        str_key = self._coerce_key(key)
+        member = self.banks_table.get_keyed_bank_member(
+            self._import_bank.bank_id, str_key
+        )
+        if member:
+            self.banks_table.remove_bank_member(member.bank_member_id)
+
+    def convert_opinions_to_label(self, opinions: t.Sequence[SignalOpinion]):
+        """
+        Simplistic 1:1 translation of opinion tag to label of same name. This
+        could be made more configurable on an API / Collab basis via the UI.
+
+        This should be an instance method even though it does not use self. This
+        is to ensure that future modifications (that require loading stuff from
+        ddb using self.banks_table) can be written.
+        """
+        return reduce(
+            lambda tags, acc: acc.union(tags), map(lambda x: x.tags, opinions)
```

Comment:
nit: You can even generator comprehension!
```suggestion
            lambda tags, acc: acc.union(tags), (o.tags for o in opinions)
```